lib/rules/method.js

```diff
@@ -99,6 +105,8 @@ function checkCallExpression(ruleHelper, callExpr, node) {
     case "CallExpression":
     case "ThisExpression":
     case "NewExpression":
+    case "TSAsExpression":
```

Comment:
I'm not sure that's correct. We need to look inside.
Example of bad code:
> lol.innerHTML = (foo as string);
Example of good code:
> lol.innerHTML = (5 as string);

(Pls let me know if my understanding of typescript is incorrect here!)

Reply:
do I misunderstand in thinking that this would be handled by `lib/rules/property.js`?  Regardless this would need updating

Reply:
My example was wrong, but my point should still be valid:
A better example of code where we'd have to look at the thing that's being do a
allowed: `insertAjacentHTML('afterned', <string>evil)`
disallowed: `insertAjacentHTML('afterned', <string>'something hardcoded')`

Reply:
added some test cases - lmk if they look good!